custom_components/sonos_alarm/switch.py

```diff
@@ -131,6 +128,13 @@ def __init__(self, soco, alarm):
             ATTR_PLAY_MODE: str(self.alarm.play_mode),
             ATTR_SCHEDULED_TODAY: self._is_today
         }
+
+        self._name = "Sonos Alarm {} {} {}".format(
+            speaker_info["zone_name"],
+            self.alarm.recurrence.title(),
+            str(self.alarm.start_time)[0:5]
+        )
+
```

Comment:
This should be easily achievable if you include those lines also in the update routine.

Reply:
You are right, I've added that now!